### PR TITLE
feat: make configurable the security context of each container part of SPOD in the SecurityProfilesOperatorDaemon cofiguration

### PR DESCRIPTION
**Pull request by ccojocar**
#### What type of PR is this?

feature

#### What this PR does / why we need it:

This PR extends the SecurityProfilesOperatorDaemon configuration to allow the security context to be specified for each container part of SPOD pod. This is required for instance to configure the SELinux options for nodes which doesn't enforce SELinux by default and do not provide the spc_t type. Such an example is the Flatcar Linux distribution.

#### Does this PR have test?

Yes 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: api/spod/v1alpha1/spod_types.go [hunk 1]
```diff
@@ -33,6 +33,14 @@ type SelinuxOptions struct {
 	AllowedSystemProfiles []string `json:"allowedSystemProfiles,omitempty"`
 }
 
+// SecurityContext define the security context of a container part of SPOD daemon pod
+type SecurityContext struct {
```
Comment (jhrozek) [inline api/spod/v1alpha1/spod_types.go:37]: Is it a good idea to name the struct the same as the one coming from `corev1` ?

**Comment by ccojocar** [comment]
> So one question that I wonder after reading the patch is you think this fine-grained configuration has uses beyond setting the SELinux context. If it's about setting the selinux context only, wouldn't it be easier to just auto-detect if the nodes support selinux (or have that configurable) rather than expose all the options?

Configuring SELinux is one issue I stumbled across, and I wasn't able to get the daemon running without changing the hard-coded value. I see that it can be useful to modify or to overwrite other security parameters such as the seccomp profile.

File: api/spod/v1alpha1/spod_types.go (hunk 1 above)
Comment (ccojocar) [reply api/spod/v1alpha1/spod_types.go:37]: I am open for a suggestion if you might think that this can create some confusion.

Comment (jhrozek) [reply api/spod/v1alpha1/spod_types.go:37]: What about making it either prefixed with `SPO` or such or naming it after the fact that it pairs the container and the contex like `SpoContSecCtx`.

**Comment by saschagrunert** [comment]
> I can just detect if the selinux is enabled and add the type if is in enforcing mode.

I would prefer that route :+1:

**Comment by ccojocar** [comment]
closing this in favour of #828

**Pull request closed without merging**